agent/xds/listeners.go

```diff
@@ -1068,14 +1056,64 @@ func (s *ResourceGenerator) makeTerminatingGatewayListener(
 		}
 	}
 
+	for _, svc := range cfgSnap.TerminatingGateway.ValidDestinations() {
+		clusterName := connect.ServiceSNI(svc.Name, "", svc.NamespaceOrDefault(), svc.PartitionOrDefault(), cfgSnap.Datacenter, cfgSnap.Roots.TrustDomain)
+
+		intentions := cfgSnap.TerminatingGateway.Intentions[svc]
+		svcConfig := cfgSnap.TerminatingGateway.ServiceConfigs[svc]
+
+		cfg, err := ParseProxyConfig(svcConfig.ProxyConfig)
```

Comment:
Hmm, I think it's a little confusing to add it to the `destination` struct since it's already in the `service-defaults` and `proxy-defaults` structs. We could make it private, but then I think it's more work to make sure it's inherited correctly.

Reply:
Per our discussion, we'll keep it as-is now and we can add this separate field later if needed.